x/nft/types/collection.go

```diff
@@ -32,13 +32,13 @@ func EmptyCollection() Collection {
 func (collection Collection) GetNFT(id uint64) (nft NFT, err sdk.Error) {
 
 	for _, nft := range collection.NFTs {
-		if nft.ID == id {
-			return nft, nil
+		if nft.GetID() == id {
+			return nil, nil
```

Comment:
@fedekunze shouldn't this return the nft?

Reply:
dah right